tests/components/QuestionStore.test.ts

```diff
@@ -76,3 +97,45 @@ test("should not return the same array", () => {
   const shuffled = shuffleArray([...questions]);
   expect(shuffled).to.not.deep.equal(questions);
 });
+
+test("Filter questions by a specific course, subject, and system", () => {
+  const filterTags = {
+    course: "VETS2011",
+    subject: "Physiology",
+    system: "Neurophysiology",
+  };
+  const filteredQuestions = filterQuestionsByTags(questions, filterTags);
+  expect(filteredQuestions.length).equal(2);
+  expect(
+    filteredQuestions.every(
+      (q) =>
```

Comment:
Similar comment to Jason, let's try to name our variables in higher order functions